### PR TITLE
#162163002 implement creating of user profiles

### PR DESCRIPTION
**Pull request by Luleherll**
#### What does this PR do?
- Creates user profiles upon registration
- Enables user to update their profile
- Writes tests for user profile endpoints
- Updates swagger documentation setup

#### Description of the tasks to be completed?
- Creates profiles app
- Updates `UserSerializer` in `authentication` app
- Updates swagger documentation setup
#### How should this be manually tested?
- Pull the branch to your remote workspace
- Run the server using `python manage.py runserver`

##### *Don't forget: Append the endpoints to the server URL*
- Post JSON data of the new user to the `api/users/` endpoint using Swagger documentation on `/docs` or POSTMAN.
```
Data format
{
  "user": {
        "email": "lule@dev.com",
        "username": "lule",
        "password": "luleadmin",
  }
}
```
- Go to `api/profiles/<username>`. Replace `<username>` with the user you have registered and you should see their profile.
```
{
  "profile": {
    "username": "lule",
    "bio": "",
    "image": null,
    "following": false
  }
}
```
- Make a `PUT` request with JSON data to the `api/user` endpoint and it should return the updated data.
```
Accepted fields
(username, email, bio, image)

Data format
{
  "user": {
    "email": "lule@dev.com",
    "token": "jwt.token.here",
    "username": "lule",
    "bio": "Just another writer",
    "image": null
  }
}
```

#### Relevant pivotal tracker stories
#162163002

### CODE REVIEW

**Comment by billkabanga** [comment]
Good work on creating the Base test class as agreed.

File: authors/apps/authentication/views.py [hunk 1]
```diff
@@ -53,18 +53,24 @@ class UserRetrieveUpdateAPIView(RetrieveUpdateAPIView):
     serializer_class = UserSerializer
 
     def retrieve(self, request, *args, **kwargs):
-        # There is nothing to validate or save here. Instead, we just want the
-        # serializer to handle turning our `User` object into something that
-        # can be JSONified and sent to the client.
```
Comment (malep2007) [inline authors/apps/authentication/views.py]: @Luleherll why are the comments being deleted?

File: authors/apps/profiles/models.py [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    updated_at = models.DateTimeField(auto_now=True)
+
+    def __str__(self):
+        return self.user.username
+
+
+# This signals the `Profile` model when a user has been registered
+# for their profile to be created.
+@receiver(post_save, sender=User)
+def create_related_profile(sender, instance, created, *args, **kwargs):
```
Comment (malep2007) [inline authors/apps/profiles/models.py]: please refactor this method. @Luleherll

File: authors/apps/profiles/serializers.py [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+class ProfileSerializer(serializers.ModelSerializer):
+    
+    username = serializers.CharField(source='user.username')
+    email = serializers.CharField(source='user.email')
+    bio = serializers.CharField(allow_blank=True, required=False)
+    
+    class Meta:
+        model = Profile
+        fields = ('username', 'email', 'bio', 'image', 'following')
+        read_only_fields = ('username',)
```
Comment (malep2007) [inline authors/apps/profiles/serializers.py]: Fix new line issues in your code base @Luleherll

File: authors/apps/profiles/urls.py [hunk 4]
```diff
@@ -0,0 +1,8 @@
+from django.urls import path
+from .views import ProfileRetrieveAPIView
+
+urlpatterns = [
+    path('profiles/<username>', ProfileRetrieveAPIView.as_view(),
+         name="get_profile"),
+    
+]
```
Comment (malep2007) [inline authors/apps/profiles/urls.py]: @Luleherll , `newline issues`

**Review by malep2007 — CHANGES_REQUESTED** [review]
@Luleherll , please address the issues in comments on this PR first

**Comment by Luleherll** [comment]
Okay, working on it now.

**Comment by aayebare** [comment]
nice work Lule. we still need to change the PR title though and add a related pivotal tracker story.
Otherwise ..great

**Review by okellogabrielinnocent — APPROVED** [review]
Good work.

**Pull request closed without merging**